app/frontend/src/components/Select.tsx

```diff
@@ -0,0 +1,79 @@
+import {
+  FormControl,
+  InputLabel,
+  Select as MuiSelect,
+  SelectProps,
+} from "@material-ui/core";
+import classnames from "classnames";
+import React from "react";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  formControl: {
+    "& .MuiOutlinedInput-root": {
+      borderRadius: theme.shape.borderRadius * 3,
+    },
+    "& .MuiInputBase-input": {
+      height: "auto",
+    },
+    display: "block",
+  },
+}));
+
+export default function Select<
+  T extends { [key: string]: string } | { [key: number]: string }
+>({
+  id,
+  children,
+  className,
+  data,
+  label,
+  variant = "outlined",
+  options,
+  ...otherProps
+}: SelectProps & {
+  id: string;
+  options?: Extract<keyof T, string | number>[];
+  value?: T extends undefined ? string | number : keyof T;
+  data?: T;
+}) {
+  const classes = useStyles();
+
+  const SelectOptions = () => {
```

Comment:
Can we lift this out of this component? Creating a component inside a component is an anti-pattern as it creates a new function reference each time so React won't know it's the same component and so screws up its reconciliation and ends up recreating the component tree and replacing more things in the DOM than necessary

Reply:
I can lift this out of the component. But I can also use useMemo for this. Any preference?

Reply:
Moving it out is better, but in this case you can just inline the options instead of having another component